sql/sql_acl.cc

```diff
@@ -14312,6 +14313,22 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
+  else if (redirect_enabled)
+  {
+    char msg[MAX_REDIRECTION_LEN];
+    size_t len = 0;
+
+    msg[0] = '\0';
+    len = snprintf(msg + 1, MAX_REDIRECTION_LEN - 1, "Location: mysql://[%s]:%s/user=%s&ttl=%s\n", 
+      redirect_server_host, redirect_server_port, sctx->user, redirect_server_ttl);
```

Comment:
why do you have a user here? Is the hypothetical client not good enough to use the  same user credentials when reconnecting to another endpoint?

Reply:
Right. Because for some cloud providers with GW or proxy as middle man, like Azure, the user name after redirection is different than the origin one. Because in Azure, part of user name tells which is the true mysql server in backend with a light proxy in machine hosts the MariaDB. So, to make sure client receive same format here and compatible with Azure, here is using same format.
For future MariaDB cloud provider with proxy as middle man, this format will be more flexible to let them define their username.